tools/automate_bindings.py

```diff
@@ -639,32 +639,40 @@ def generateMethodsArray(module_name, methods):
     DECLARE_CLASS(
         %(module_name)s,
         L,
-        wrap<luawt_%(module_name)s_make>::func,
+        %(make)s,
         0,
         luawt_%(module_name)s_methods,
         base
     );
 }
 '''
 
-def generateModuleFunc(module_name, base):
+def generateModuleFunc(module_name, base, is_not_abstract):
     base = base.name
+    if is_not_abstract:
+        make = 'wrap<luawt_%s_make>::func' % module_name
+    else:
+        make = '0'
     options = {
         'module_name' : module_name,
+        'make' : make,
         'base' : base,
     }
     return MODULE_FUNC_TEMPLATE.lstrip() % options
 
 def generateConstructor(module_name, constructors):
-    constructor_name = 'make'
-    constructor_return_type = module_name + ' *'
-    return implementLuaCFunction(
-        True,
-        module_name,
-        constructor_name,
-        [c.arguments for c in constructors],
-        constructor_return_type,
-    )
+    if len(constructors) != 0:
```

Comment:
1. Use [Guard Clause](http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) here.
2. You can write just `if constructors:`